### PR TITLE
Update validation and action with respect to require_master_price config

### PR DESCRIPTION
**Pull request by tanmay3011**
Please review last 2 commits... once vinsol/spree:master is updated with respect to spree/spree:master all commits not related to this PR will automatically get removed.

PLEASE DO NOT MERGE... Close this PR once reviewed will create a direct PR with respect to spree/spree:master once accepted

### CODE REVIEW

File: core/app/models/spree/variant.rb [hunk 1]
```diff
@@ -35,8 +35,8 @@ class Variant < Spree::Base
       inverse_of: :variant
 
     before_validation :set_cost_currency
+    validate :check_price, if: proc { Spree::Config[:require_master_price] }
```
Comment (priyank-gupta) [inline core/app/models/spree/variant.rb]: @tanmay3011 Remove extra line

**Comment by tanmay3011** [comment]
@priyank-gupta fixed!

**Comment by priyank-gupta** [comment]
:+1:
Also, it would be nice if we add specs for `#check_price`.

**Comment by tanmay3011** [comment]
https://github.com/spree/spree/pull/6967

**Pull request merged**